pkg/controller/controller.go

```diff
@@ -182,8 +186,57 @@ func (c *controller) Run(workers int, stopCh <-chan struct{}) {
 		if utilfeature.DefaultFeatureGate.Enabled(scfeatures.AsyncBindingOperations) {
 			createWorker(c.bindingPollingQueue, "BindingPoller", maxRetries, false, c.requeueServiceBindingForPoll, stopCh, &waitGroup)
 		}
+
+		//createWorker(c.ClusteIDConfigMapQueue, "ClusteIDConfigMap", maxRetries, true, c.reconcileClusteIDConfigMapKey, stopCh, &waitGroup)
 	}
 
+	func() {
+		waitGroup.Add(1)
+		// Cannot wait for the informer to push something into a queue.
+		// What we're waiting on may never exist without us configuring
+		// it, so we have to poll/ ask for it the first time to get it set.
+
+		// Can we ask 'through' an informer? Is it a writeback cache? I
+		// only ever want to monitor and be notified about one configmap
+		// in a hardcoded place.
+		go func() {
+			wait.Until(func() {
+				glog.V(9).Info("cluster ID monitor loop enter")
+				cm, err := c.kubeClient.CoreV1().ConfigMaps("default").Get("cluster-info", metav1.GetOptions{})
+				if errors.IsNotFound(err) {
+					m := make(map[string]string)
+					m["id"] = string(uuid.NewUUID())
+					cm := &corev1.ConfigMap{
+						ObjectMeta: metav1.ObjectMeta{
+							Name: "cluster-info",
+						},
+						Data: m,
+					}
+					c.kubeClient.CoreV1().ConfigMaps("default").Create(cm)
+					// if we fail to set the id,
+					// it could be due to permissions
+					// or due to being already set while we were trying
+					glog.Warning("could not set clusterid configmap", cm)
+				} else {
+					// cluster id exists and is set
+					// get id out of cm
+					if id, ok := cm.Data["id"]; ok {
+						// set it if it has not been set
+						if "" == c.clusterID {
+							c.clusterID = id
+						} else if id != c.clusterID {
+							glog.Warningf("got a different cluster id than what was stored. controller had %q, got %q.", c.clusterID, id)
+						}
+					} else {
+						glog.Warning("got a clusterid configmap, but it had no id")
```

Comment:
we should probably set the value in this case. I can't think of why we would treat it differently than the CM missing.

Reply:
I am not inclined to make state changes that nobody has asked for.

Reply:
If someone erased the "id" key but not the entire map then we need to recreate it.  It should be treated the same as a missing map. Either way we need to get back to a good state.  A missing "id" isn't a good state.

Reply:
I guess I agree anyway.